build.rs

```diff
@@ -47,6 +51,23 @@ fn main() {
         Triple::from_str(&target).unwrap_or_else(|_| panic!("Invalid target name: '{}'", target));
     let out = File::create(out_dir.join("host.rs")).expect("error creating host.rs");
     write_host_rs(out, triple).expect("error writing host.rs");
+    if using_less_1_40().is_none() {
+        println!("cargo:rustc-cfg=feature=\"rust_1_40\"");
+    }
+}
+
+fn using_less_1_40() -> Option<()> {
```

Comment:
Could this return a `bool` return than an `Option<()>`? And could the sense be inverted, as in `fn using_at_least_1_40()` rather than `less` so that it doesn't need to be negated (`is_none()`) above?

Reply:
Sure, I was trying to be clever and avoid `match {` everywhere but having it make sense is probably more important than making it concise.

Reply:
Ok I was still a _little_ clever, let me know if you want me to get rid of the immediately executed closure.

Reply:
Ah, I didn't notice that there were `?`s taking advantage of the `Option` return type. That `match` trick is still a little tricky, but the main thing is that the code in `main` reads easily, so this looks good to me.